src/main/database.cpp

```diff
@@ -21,6 +22,20 @@ using namespace kuzu::transaction;
 namespace kuzu {
 namespace main {
 
+void getFileFlags(SystemConfig systemConfig, int& flags, FileLockType& lock, bool create_new) {
```

Comment:
1. Simply pass in `accessMode` is enough, no need to pass in the whole SystemConfig.
2. Make this function static.
3. Rename to `getLockFileFlagsAndType`.
4. Reorder your params, so your input and output are separate and together. better for reading. for example `getFileFlags(SystemConfig systemConfig, bool create_new, int& flags, FileLockType& lock)`